lib/exercism/metrics.rb

```diff
@@ -0,0 +1,45 @@
+require 'benchmark'
+require 'socket'
+
+class Metrics
+  VALID_METRIC_NAME = /\A[a-z_.]+\z/i
+
+  attr_reader :api_key, :host, :port
+
+  def self.time(metric_name, &block)
+    new.time(metric_name, &block)
+  end
+
+  def self.increment(metric_name)
+    new.increment(metric_name)
+  end
+
+  def initialize(api_key: ENV['HOSTEDGRAPHITE_APIKEY'], host: 'carbon.hostedgraphite.com', port: 2003)
```

Comment:
No, it was an intentional choice to make `Metrics` no-op when the API key is not present. Consider that most everyone working with this project in development will not have an API key. I don't want to force a dummy key to be provided or an account to be set up. In most cases, developers won't care to look at the metrics anyway.

Now that you mention this though, I just thought of something… instead of doing nothing, `Metrics` can just print to stdout instead of doing nothing.